TYPO3.Neos/Classes/TYPO3/Neos/Routing/FrontendNodeRoutePartHandler.php

```diff
@@ -384,19 +398,16 @@ protected function resolveRoutePathForNode(NodeInterface $node)
     protected function getRelativeNodePathByUriPathSegmentProperties(NodeInterface $siteNode, $relativeRequestPath)
     {
         $relativeNodePathSegments = [];
+        $documentNodeType = $this->nodeTypeManager->getNodeType('TYPO3.Neos:Document');
+        $context = $siteNode->getContext();
         $node = $siteNode;
 
         foreach (explode('/', $relativeRequestPath) as $pathSegment) {
-            $foundNodeInThisSegment = false;
-            foreach ($node->getChildNodes('TYPO3.Neos:Document') as $node) {
-                /** @var NodeInterface $node */
-                if ($node->getProperty('uriPathSegment') === $pathSegment) {
-                    $relativeNodePathSegments[] = $node->getName();
-                    $foundNodeInThisSegment = true;
-                    break;
-                }
-            }
-            if (!$foundNodeInThisSegment) {
+            $nodes = $this->nodeSearchService->findByProperties(['uriPathSegment' => $pathSegment], [$documentNodeType], $context, $node);
```

Comment:
Anyways, I've added a filter for that. Could be nice if there was a parameter to `findByProperties` to only consider direct children, WDYT?